lib/spack/spack/cmd/module.py

```diff
@@ -102,17 +101,54 @@ def setup_parser(subparser):
     )
 
 
-class MultipleMatches(Exception):
-    pass
+class MultipleSpecsMatch(Exception):
+    """Raised when multiple specs match a constraint, in a context where
+    this is not allowed.
+    """
```

Comment:
I think just a `docstring` is fine either. We should have errors coming from the interpreter otherwise.